src/RewardsDistribution.sol

```diff
@@ -0,0 +1,201 @@
+     */
+    event File(bytes32 indexed what, uint256 data);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. Currently the supported values are: "calc"
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice A distribution of tokens was made.
+     * @param amount The total tokens in the current distribution.
+     */
+    event Distribute(uint256 amount);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RewardsDistribution/not-authorized");
+        _;
+    }
+
+    /**
+     * @dev The token `gem` used in DssVest must be the same as `rewardsToken` in StakingRewards.
+     * @param _dssVest The DssVest instance as the source of the funds.
+     * @param _stakingRewards The farming contract.
+     * @param _calc The contract the function to calculate how the rewards distribution must be done.
+     */
+    constructor(address _dssVest, address _stakingRewards, address _calc) {
+        require(
+            DssVestWithGemLike(_dssVest).gem() == StakingRewardsLike(_stakingRewards).rewardsToken(),
+            "RewardsDistribution/invalid-gem"
+        );
+
+        dssVest = DssVestWithGemLike(_dssVest);
+        stakingRewards = StakingRewardsLike(_stakingRewards);
+
+        setCalc(_calc);
+        emit File("calc", _calc);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Administration ---
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"vestId"
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "vestId") {
+            setVestId(data);
+        } else {
+            revert("RewardsDistribution/file-unrecognized-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /**
+     * @notice Updates the `vestId` managed by this contract.
+     * @dev The vest must be valid, in favor of `stakingRewards` and managed by this contract.
+     * @param _vestId The new vest ID.
+     */
+    function setVestId(uint256 _vestId) internal {
+        require(dssVest.valid(_vestId), "RewardsDistribution/invalid-vest-id");
+        require(dssVest.usr(_vestId) == address(stakingRewards), "RewardsDistribution/invalid-vest-usr");
+        require(dssVest.mgr(_vestId) == address(this), "RewardsDistribution/invalid-vest-mgr");
```

Comment:
Gotcha... I believe I overlooked the actual implementation a bit too much.

This is actually an argument in favor of importing the real dependencies instead of mocking, because I'd probably build the mock with the wrong assumption.

Thanks for pointing this out.

Reply:
Fixed. Can't resolve comments.